Feature interval timer/euihyeok

thesoundfromthesky opened the pull request:
1. Changed function complete_interval() to calculate the elapsed time of the last completed interval.
2. on_complete_interval introduced as first constructor parameter of class IntervalTimer.
3. Refactored overall logic

BrannanC commented on interval-timer.js:
```diff
@@ -0,0 +1,88 @@
+  }
+
+  retrieve_all_completed_intervals() {
+    this.completed_intervals.forEach((e, i) =>
```
Let's make this return the completed intervals array instead of console.log

thesoundfromthesky replied:
Resolved it by making function retrieve_all_completed_intervals() to return the completed intervals array

BrannanC commented on interval-timer.js:
```diff
@@ -0,0 +1,88 @@
+    }
+  }
+  
+  complete_interval() {
```
I think we should do the check for elapsed time here, otherwise we'll run into some weird problems. We should also be checking for is_running and if it's paused.

thesoundfromthesky replied:
Null check for variable elapsed_time and is_running added to function complete_interval before invoking.

thesoundfromthesky commented:
1. Moved IntervalTimer to main.js
2. Removed test code
3. Rollbacked readme.md to original
4. Made retrieve_all_completed_intervals() return the completed intervals array instead of console.log
5. Added null check before invoking complete_interval()

> And let me know if you have any ideas for how we should do testing.

I was thinking between Jasmine or Jest for unit testing, but I would probably go with Jest. How would you like to do unit testing?
